### PR TITLE
Update the bootstrap compiler

### PR DESCRIPTION
**Pull request by alexcrichton**
Now that we've got a beta build, let's use it!

### CODE REVIEW

**Comment by brson** [comment]
This is still waiting on a beta that contains a cargo fix for https://github.com/rust-lang/cargo/pull/3280. Next beta after https://github.com/rust-lang/rust-packaging/pull/66 should have it.

**Comment by brson** [comment]
Subsumes https://github.com/rust-lang/rust/pull/37635/files

**Comment by alexcrichton** [comment]
Updated the Cargo date, should be good to go

File: src/librustc_llvm/lib.rs [hunk 1]
```diff
@@ -24,9 +24,9 @@
 
 #![feature(associated_consts)]
 #![feature(box_syntax)]
+#![feature(concat_idents)]
 #![feature(libc)]
 #![feature(link_args)]
-#![feature(staged_api)]
```
Comment (eddyb) [inline src/librustc_llvm/lib.rs]: This can't/shouldn't be removed - Travis failed on this.

File: src/tools/compiletest/Cargo.toml [hunk 2]
```diff
@@ -4,12 +4,6 @@ name = "compiletest"
 version = "0.0.0"
 build = "build.rs"
 
-# Curiously, this will segfault if compiled with opt-level=3 on 64-bit MSVC when
-# running the compile-fail test suite when a should-fail test panics. But hey if
-# this is removed and it gets past the bots, sounds good to me.
```
Comment (eddyb) [inline src/tools/compiletest/Cargo.toml]: This comment should be preserved, or `opt-level = 2` be removed otherwise at the top-level.

**Review by eddyb — CHANGES_REQUESTED** [review]
r=me with comments addressed.

**Pull request merged**